src/raftstore/store/region_snapshot.rs

```diff
@@ -23,22 +24,29 @@ use tikv_util::{panic_when_unexpected_key_or_data, set_panic_mark};
 ///
 /// Only data within a region can be accessed.
 #[derive(Debug)]
-pub struct RegionSnapshot {
-    snap: RocksSyncSnapshot,
+pub struct RegionSnapshot<E: KvEngine> {
+    snap: <E::Snapshot as Snapshot>::SyncSnapshot,
     region: Arc<Region>,
     apply_index: Arc<AtomicU64>,
 }
 
-impl RegionSnapshot {
-    pub fn new(ps: &PeerStorage) -> RegionSnapshot {
+impl<E> RegionSnapshot<E>
+where
+    E: KvEngine,
+{
+    #[allow(clippy::new_ret_no_self)] // temporary until this returns RegionSnapshot<E>
+    pub fn new(ps: &PeerStorage) -> RegionSnapshot<RocksEngine> {
         RegionSnapshot::from_snapshot(ps.raw_snapshot().into_sync(), ps.region().clone())
     }
 
-    pub fn from_raw(db: Arc<DB>, region: Region) -> RegionSnapshot {
-        RegionSnapshot::from_snapshot(RocksSnapshot::new(db).into_sync(), region)
+    pub fn from_raw(db: Arc<DB>, region: Region) -> RegionSnapshot<RocksEngine> {
```

Comment:
Will you change RocksEngine as KvEngine next PR?

Reply:
Probably not the next PR, but eventually yes.